### ISSUE TITLE
method call annotations for specifying value & gas.

### ISSUE THREAD

**Issue by gavofyork** [ethereum/cpp-ethereum#686]
This should work:

```
contract Bar { function bar(uint _a) {} }
contract Foo {
  function foo(address a) {
    Bar(a) b;
    b.bar{value: 3 ether, gas: 1500}(50);
  }
}
```

Also, there should be `gas: all` and `gas: max` (determines it from ABI JSON or Solidity code).

**Comment by chriseth** [comment]
Two alternative suggestions that would not require such a drastic change in the syntax:
1.: ```b.bar.options({value: 3 ether, gas: 1500})(50);```
2.: ```b.bar.value(3 ether).gas(1500)(50);```, ```b.bar.gasMax().gasAll()(50)```.

**Comment by chriseth** [comment]
Addressing only specific gas values for now until we have a way to measure gas usage.

**Comment by chriseth** [comment]
Implemented in #799.
